R/http.R

```diff
@@ -354,6 +326,58 @@ queryString <- function(elements) {
   return(result)
 }
 
+buildPath <- function(apiPath, path, query = NULL) {
+  # prepend the service path
+  url <- paste(apiPath, path, sep = "")
+
+  # append the query
+  if (!is.null(query)) {
+    # URL encode query args
+    query <- utils::URLencode(query)
+    url <- paste(url, "?", query, sep = "")
```

Comment:
it surprises me that we construct our own query string and concatenate it with the URL, but maybe that's necessary because the different transports don't all accept query lists in similar ways.